.editorconfig

```diff
@@ -88,9 +88,9 @@ dotnet_remove_unnecessary_suppression_exclusions = none
 [*.cs]
 
 # var preferences
-csharp_style_var_elsewhere = false:silent
```

Comment:
So, my experience from making things errors in other projects is that it becomes super annoying during local dev.

Like we're making it impossible to use `var` while hacking on things.

Can we make these warn and then rely on TreatWarningsAsErrors in CI to prevent it from getting checked in?

I know this is a slight reversal from my position a while ago, but I think I'm landing on "style" should be enforced in CI and easily fixable locally, but not block builds.

I think warn is the easiest way to do that.

Reply:
I think it's reasonable. We can change it either way if it gives us trouble.